### DISCUSSION TITLE
FetchError only on Vercel

### DISCUSSION THREAD

**Discussion by markwongwkh-code** [vercel/community#623] [Help] [answered]
What i am doing is creating a basic CRUD blog app. I can run locally without any problems. But after I deploy to Vercel, the system keeps having the below error:

FetchError: invalid json response body at (link) reason: Unexpected token < in JSON at position 0

I really don't know whether this is a problem about React version or not. I can run with React 17 but not on React 18. All environment variables should be correct.

Can anyone please give me a helping hand?

Thank you!

**Comment by gogocharli** [comment, marked as answer]
>  I can run with React 17 but not on React 18

Have you tried updating Next.js to the latest version when using React 18? I noticed in your `package.json` file that you're using Next 11.1.0.

**Reply by jbarrasso** [reply]
I'm having a similar issue which is being discussed here as well - requests work fine in local environment but return 404s when deployed.
https://github.com/vercel/community/discussions/604#discussioncomment-2817708

Could you share what your request and responses headers look like in the local and deployed environments? 

(Inspect the browser web page > Network tab > Fetch/HXR > click on the failed request > Headers

Thanks

**Comment by jbarrasso** [comment]
Have you made any progress on this? My similar issue is still not resolved

**Reply by markwongwkh-code** [reply]
> Have you made any progress on this? My similar issue is still not resolved

Turnout I found the reason is my environment variable issue. Make sure your variables match your settings.
